src/SDK/SmartDetectors.Extensions/AlertExtensions.cs

```diff
@@ -195,23 +246,29 @@ private static AlertProperty CreateAlertProperty(Alert alert, PropertyInfo prope
         /// <param name="propertyName">The table property name.</param>
         /// <param name="displayName">The table property display name.</param>
         /// <param name="tableAttribute">The attribute applied to the table property.</param>
+        /// <param name="order">The order to use</param>
         /// <returns>The newly created <see cref="TableAlertProperty{T}"/> instance.</returns>
         private static DisplayableAlertProperty CreateTableAlertProperty(
             object propertyValue,
             string propertyName,
             string displayName,
-            TablePropertyAttribute tableAttribute)
+            TablePropertyAttribute tableAttribute,
+            Order order)
         {
             // Validate we have a proper value
             if (!(propertyValue is IList tablePropertyValue))
             {
                 throw new ArgumentException("An AlertPresentationTableAttribute can only be applied to properties of type IList");
             }
 
-            Type tableRowType = GetGenericListType(propertyValue.GetType());
-            if (tableRowType == null)
+            // Get element type, and verify that all elements are of the same type
+            Type tableRowType = tablePropertyValue[0].GetType();
+            foreach (object item in tablePropertyValue)
```

Comment:
>foreach (object item in tablePropertyValue) [](start = 12, length = 43)

please change to if (!tablePropertyValue.All(...))

Reply:
Since it's an IList, I can't Linq it - tried a few alternatives, this one seemed the clearest to me

---
In reply to: [254177544](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/45#discussion_r254177544) [](ancestors = 254177544)